### PR TITLE
Clarify definition of ansible.builtin.success

### LINKED ISSUE #83698: ansible.builtin.success reports that skipped tasks completed successfully
### Summary

The `ansible.builtin.success` test reports true for skipped tasks.

Even if this is somehow the intended behaviour it is still a serious documentation bug.
If not doing a task at all is somehow meant to be interpreted as completing a task successfully then any reference to "completed", "finished", or "success" should be removed from [the docs](
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/success_test.html).

### Issue Type

Bug Report

### Component Name

ansible.builtin.success

### Ansible Version

```console
$ ansible --version
ansible [core 2.16.9]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/fedora/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.12/site-packages/ansible
  ansible collection location = /home/fedora/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.2 (main, Feb 21 2024, 00:00:00) [GCC 14.0.1 20240217 (Red Hat 14.0.1-0)] (/usr/bin/python3)
  jinja version = 3.1.3
  libyaml = True


$ rpm -q ansible
ansible-9.8.0-1.fc40.noarch
```

### Configuration

```console
$ ansible-config dump --only-changed -t all
CONFIG_FILE() = /etc/ansible/ansible.cfg
EDITOR(env: EDITOR) = /usr/bin/nano
```

### OS / Environment

Fedora 40

### Steps to Reproduce

```yaml (paste below)
---
- hosts: localhost
  tasks:
    - name: Do something really important but it might be skipped
      ansible.builtin.reboot:
      register: thing
      when: false

    - name: Print result
      debug:
        msg: "Actually that really important thing definitely completed successfully"
      when: thing is ansible.builtin.success

```

### Expected Results

The task is skipped. It did not complete successfully so the `success` test must not return true.

### Actual Results

```console
$ ansible-playbook test.yml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************
ok: [localhost]

TASK [Do something really important but it might be skipped] ****************************************************************************************************************
skipping: [localhost]

TASK [Print result] *********************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Actually that really important thing definitely completed successfully"
}

PLAY RECAP ******************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   
```
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by rptaylor**
Explain the danger of assuming that "success" means actual completion

##### SUMMARY

See https://github.com/ansible/ansible/issues/83698
Closes https://github.com/ansible/ansible/issues/83698 
But more importantly this Fixes https://github.com/ansible/ansible/issues/83698  which was already closed without any solution.

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

See linked issue.

### CODE REVIEW

File: lib/ansible/plugins/test/success.yml
Comment (jborean93) [inline lib/ansible/plugins/test/success.yml:8]: I think this line should probably stay, we can clarify like you've done later on that a skipped result is also considered a success.

**Review by bcoca — CHANGES_REQUESTED** [review]
I would not add the 'or skipped'/'skipped or' nor delete the first line in the description.

This should just add a new (third)line like:
` Tasks that are skipped, have not failed, hence they are considered a success in the context of this test`

**Comment by rptaylor** [comment]
Thanks for the feedback. I have tried to address it without further highlighting the contradiction/ambiguity between different interpretations of "success" and "complete".

IMHO the purpose of the documentation should be to clearly and directly explain to users in common language what the test does, not to hew to technicalities about how a task can be interpreted as "successful" and "complete" from a programming point of view (e.g. based on how the ansible engine parsed the task) even though it did NOT succeed or complete on a host from the actual task execution point of view that a user cares about.

If you still feel that the `success` test should be described as returning true if a task "completed (or finished) successfully", I think it would be necessary to precisely define what "complete" and "success" mean to avoid contradictions and ambiguity.

**Comment by sivel** [comment]
I feel like the line you removed explained it.  The `success` test is the opposite of `failed`.  It itself only tests to see if the task is `failed`, and if not, it is considered sucessful.  In effect `success` == `not failed`.

**Comment by rptaylor** [comment]
@sivel  Yes, I did not remove the first line but would change it from "finished successfully" to "did not fail". I think we are in agreement that this is precise, consistent and correct, based on the definition you stated.

If we are talking about precise technical definitions, `success` is equivalent by definition to, and interchangeable with, `not failed`, so what I have written is logically equivalent to the previous text. It is equally correct now so there should not be an issue with it, right?

If we're talking about what people expect the word "success" of a task to mean, it should clarify that it can mean the task execution _did not run on the host_ and _did not succeed on the host_.

The simplest way I can think of to explain this is to say it "succeeded or was skipped". 
Alternatively it could say "Tests if task finished successfully being processed by Ansible (regardless of whether or not the task was executed or successful on the host)" but that is longer and likely to raise more questions.

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [inline lib/ansible/plugins/test/success.yml]: I  think the original phrasing reads better and is equivalent in meaning.

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [inline lib/ansible/plugins/test/success.yml]:
```suggestion
    - A skipped task is a successful task. See P(ansible.builtin.skipped#test) to test if a task is skipped.
```

Though this is redundant still with the line above - if a task is skipped, it won't contain a 'failed' key in the input dictionary.

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: I was looking at the other issue, and I think this stems from a misunderstanding between tasks and actions. If you run a single action and it's skipped, loop over a task with an empty list, or all of the actions in the loop are skipped, then the task result is `ansible.builtin.skipped`. If you run a single action and it fails, or any action in the loop fails, then the overall task result is failed (and success is the inverse of that, which includes skipped as a subset).

```suggestion
    - To validate if the task executed all actions, see P(ansible.builtin.skipped#test).
```

File: lib/ansible/plugins/test/success.yml
Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: If you propose to say that the task "finished successfully", there would need to be heavy qualifications about what those words mean, explaining how it can be that a task that "finished successfully" did not start running on the host, therefore it could not have finished running on the host, and it could not have succeeded on the host, because it was skipped.

@s-hertel in the case of a skipped task, when nothing has been executed on the target host, can you explain exactly what thing/event/process finished successfully? Do we want to say e.g. that it "finished successfully being processed by the controller host", to draw a clear distinction against task execution on the target host?

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: Success is the opposite of failed. A dictionary is considered 'failed' when it contains the key 'failed' and the value is True. A skipped task never contains the key 'failed', therefore a skipped task is successful, no qualifier needed.

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: All skipped tasks are successful but not all successful tasks are skipped. You were right in the previous issue, that the correct pattern to check if a host executed the task is with the success + skipped tests. I think adding something like https://github.com/ansible/ansible/pull/83705#pullrequestreview-2213716186 or my suggestions below would help clarify, without requiring rewording this line.

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: If you had the assumption action == task, then I can understand why it might not be intuitive that a skipped action is success (skipped, success, and failed could have no overlap at all, for example). (In which case success would no longer necessarily be the opposite of failed)

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: @s-hertel That is a circular definition. I asked, in the case of a skipped task, what precise event or occurrence completes or is successful on a host, and you said effectively that it succeeds because it is `ansible.builtin.successful`. In other words, "`ansible.builtin.success` tests if task returns `ansible.builtin.success`".  A circular definition, of course, does not define anything.

Aside from a clear technical definition (i.e. "returns the opposite of failed"), the definition should provide an explanation in plain simple language that can be easily interpreted by end users using the established meaning of basic English words like [success](https://www.dictionary.com/browse/success) or [complete](https://www.dictionary.com/browse/complete). E.g. "the task ran on a host successfully or was skipped" or "processing of the task on the Ansible controller host successfully completed without an error".

So if you propose to say that `ansible.builtin.success` tests if a task finished successfully, can you please explain what precise event or occurrence finishes and is successful in the case of a skipped task, which does not start, run, or succeed on a host, using the standard English meaning of "finished" and "successful", without resorting to a circular definition?   Then I will be glad to write the documentation to clarify this based on your explanation.

I don't see how the distinction between a task and action is material. 

> therefore a skipped task is successful, no qualifier needed.

The issue is basically considering `ansible.builtin.success` as interchangeable with [success](https://www.dictionary.com/browse/success), when in practice the result of `ansible.builtin.success` is  counterintuitive and at odds with a common sense expectation of what "successfully completed" means in the case of a skipped task - if it successfully completed it should have executed the operation on the target host and ensured that the result was as expected (i.e. the green or orange colour coded output indicating successful completion, not teal or red indicating skipped or failed). The meanings of the English words "success" and "complete" are not really open to different interpretations, so to avoid ambiguity, contradictions, and misinterpretations, if those words are used in the documentation they should be carefully and precisely qualified to clarify what exactly has succeeded and completed and how.

Comment (bcoca) [reply lib/ansible/plugins/test/success.yml]: simple: the task did not end in failure

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: @bcoca Great, sounds good to me, we shall let the documentation say "Test if task did not end in failure". 
That is accomplished with this change I have proposed here,  although I can change it from "did not fail" to "did not end in failure" if you prefer.

File: lib/ansible/plugins/test/success.yml
Comment (bcoca) [reply lib/ansible/plugins/test/success.yml]: i find the original better, your change seems redundant

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: Not sure which change that is referring to but I agree that referring to the skipped test is useful, I updated that in line with s-hertel's suggestion. Thanks.

File: lib/ansible/plugins/test/success.yml
Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml:8]: This is covered in the other thread: https://github.com/ansible/ansible/pull/83705#discussion_r1706136358

**Comment by rptaylor** [comment]
I updated the PR with the proposed wording based on maintainer's feedback.

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: @rptaylor Register a skipped task and debug the registered variable - you will see it is a dictionary which does not contain 'failed'. If it's not failed, it's successful. Not circular, it's literally how it's written in the code.

The way you've phrased is redundant. Just delete the first part of the sentence instead of saying the same thing twice and using an unnecessary semicolon. (Though, I prefer the original)

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [inline lib/ansible/plugins/test/success.yml]: In the spirit of clarifying overloaded terms, C(failed) should really be a link to the failed test (since it's the test's definition of failed).

```suggestion
    - Tests if task finished successfully, opposite of P(ansible.builtin.failed#test).
```

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [inline lib/ansible/plugins/test/success.yml]: I think 'whether the task was executed' is inaccurate. A task may be executed, but the conditional for _modules_ it would execute may be False (skipped). A task may be the execution of multiple different actions/modules or none. The overall task only gives a status of skipped if all the items are skipped. A successful task indicates ansible-core didn't fail during any of the stages of a task (preprocessing/postprocessing/conditional module execution).

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: By the way, there is a documentation issue open about better documenting the lifecycle of a task https://github.com/ansible/ansible-documentation/issues/1295. Tasks really do a lot more than just executing something on the target.

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: @s-hertel Okay, if not "executed" then what word would you use to describe a task that is not skipped and actually resulted in operations occurring on the target host?

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: What about "To determine whether the task executed operations on the target host(s) or was skipped" ?
That indicates that the operations (of a module) are what is executed, and the task is what causes them to be executed.

File: lib/ansible/plugins/test/success.yml
Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: @s-hertel Yes, the technical details are clear, but the purpose of documentation is to explain in plain clear language how the application works so that you don't need to read the code.

> If it's not failed, it's successful. Not circular, it's literally how it's written in the code.

The crux of this issue is semantics - a skipped task is [successful](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/success_test.html) but it is not [successful](https://www.dictionary.com/browse/successful). Simply writing a word in code does not necessarily imbue the code with the meaning of that word. When there is a disconnection between the two it is especially important to highlight in documentation.

I will follow up with a revision after I hear back in the other thread. Thanks!

File: lib/ansible/plugins/test/success.yml
Comment (bcoca) [reply lib/ansible/plugins/test/success.yml]: Not all tasks will operate ON the target, the task is normally 'for the target' (there are even exceptions to this), but that does not mean it runs ON it, see actions like debug, add_host, imports/includes/etc  or most modules that rely on net_cli/netconf/httpapi plugins.

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: A skipped task is [successful](https://www.merriam-webster.com/dictionary/successful). A [task](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/task.py#L46) is an ansible-core construct that interacts with a bunch of other ansible-core constructs. A skipped task is treated as successful during the ansible runtime - it doesn't trigger rescue blocks or max_fail_percentage, or show up as an error in your play recap or give a non-zero rc.

File: lib/ansible/plugins/test/success.yml
Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: @rptaylor I gave a couple above, and used the word "executed" in the second.

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: Okay great, so we're back to 'executed'. Shall we close this thread then and keep the proposed change as it is?
"To determine whether the task was executed or skipped, see P(ansible.builtin.skipped#test)."

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: A skipped task *is* executed, so what you've written is incorrect. +1 to close the pr, thread seems to be a waste of time.

Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: I asked
> if not "executed" then what word would you use to describe a task that is not skipped

You said
> I gave a couple above, and used the word "executed" in the second.

I'm not sure what couple words above you were referring to but I thought you were suggesting "executed" after all in response to my question.

> A skipped task _is_ executed

Okay .... that sounds like another mind-boggling contradiction of overloaded/ambiguous terms that would need to be carefully detangled in documentation but I'll leave that aside for now. I'm really just looking for _a single acceptable word_ that can be used to describe a task that is not skipped. Is there any possible way to just come up with a word like that? What about "ran"?

Comment (s-hertel) [reply lib/ansible/plugins/test/success.yml]: If you get a task result, the task has executed. I think you mean, "[task](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Task)'s [action](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Action)" when you say "[task](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Task)".

The word "executed" is fine, as is "ran" or "finished" but the "skipped" test isn't capable of checking if a task executed, only the actions.

Comment (bcoca) [reply lib/ansible/plugins/test/success.yml]: ok, so i think this is a disagreement with what 'executed' means.

In task context that we are using, it is that the task was templated, processed and evaluated
I'm assuming you mean that the 'module was run on the remote'

Hence the disagreement, lets just stay away from that term and keep the current 'finished'

File: lib/ansible/plugins/test/success.yml
Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: Good idea, I added a link. Thanks!

File: lib/ansible/plugins/test/success.yml
Comment (rptaylor) [reply lib/ansible/plugins/test/success.yml]: Okay I have updated it to refer to the task's actions.

**Comment by s-hertel** [comment]
What do you think about https://github.com/ansible/ansible/pull/83766? I think we can clarify without reworking the whole description.

**Comment by rptaylor** [comment]
> What do you think about #83766? I think we can clarify without reworking the whole description.

I think everything that I said in [this thread](https://github.com/ansible/ansible/pull/83705/files/2f1e4b9cef259950896108432d6841c4bce1318c#diff-062aa00741cf691b8e035843a7eb59e261ca1c99b0a66c3a9a56c1118cea927e) , and that going with ["did not end in failure"](https://github.com/ansible/ansible/pull/83705#discussion_r1706128500) as proposed by bcoca is the most accurate and least misleading way to describe it.

**Comment by s-hertel** [comment]
I still disagree as per the thread. Your arguments haven't been logically sound, which is fine, but being dead set that your solution is the only solution is not productive.

The consensus from all reviewers has been to revert the changes to the existing documentation and add instead, so that's what I've tried to do in https://github.com/ansible/ansible/pull/83766.

By your logic you should also have a problem with the failed test documentation, as it starts by defining failure the same way the success test is defining success. 

If you don't like the phrasing regarding non-failed tasks in my PR, it is also copied from the failed test.

**Comment by rptaylor** [comment]
@s-hertel Oh, can you point out which part is not logically sound? 

I've actually tried to be rather flexible, and balance feedback that is at times conflicting between different reviewers, and have made several revisions based on the feedback. I'm open to several different approaches. However I do have a firm opinion about what words mean in common usage based on their established definitions, and when words are used in a technical context that is tricky to reconcile with common usage it makes communication difficult and documentation unclear.

I have not considered the failed test documentation and do not plan to offer any proposed changes to it at this time. 

My approach to the various semantic pitfalls (all of which I understand the technical reasons and meaning of, but the point I'm trying to get at is how to address in documentation the incongruity between the precise technical meaning and the common sense interpretation of something like e.g. ["a skipped task is executed"](https://github.com/ansible/ansible/pull/83705#discussion_r1712007472)) has been to try to eliminate them or clearly address them head on. If you want to take a different approach I guess that is up to you. 

In any case I think this illustrates the challenges of trying to have a productive clarifying discussion if we somehow do not have the same definition of what basic words like "successful" and "executed" mean as a starting point for communication (the words themselves, not to be confused with their interpretation of a specific technical concept in a technical context).

**Comment by s-hertel** [comment]
> @s-hertel Oh, can you point out which part is not logically sound? (Again I think this is really about semantics, not logic.)

I [did](https://github.com/ansible/ansible/pull/83705#discussion_r1711751631).

> I have not considered the failed test documentation and do not plan to offer any proposed changes to it at this time.

It's ironic since you deleted the line that saves users the effort of looking up the documentation for the failed test in order to understand the success test:

```diff
-  - This test checks for the existence of a C(failed) key in the input dictionary and that it is V(False) if present
```

> I've actually tried to be rather flexible, and balance feedback that is at times conflicting between different reviewers, and have made several revisions based on the feedback.

Multiple reviews can be overwhelming, but in this case I think they're in agreement. I agree fully with the reviews before me, and if the first/second/third review was addressed rather than ignored/rejected, I wouldn't have added my 2c to the PR.

> In any case I think this illustrates the challenges of trying to have a productive clarifying discussion if we somehow do not have the same definition of what basic words like "successful" and "executed" mean as a starting point for communication (the words themselves, not to be confused with their interpretation of a specific technical concept in a technical context).

We *are* using the same definition for "successful", the adverb/adjective is not the issue here. You are/were confused over tasks and actions. An [executing task]( https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/process/worker.py#L177-L187) can [execute an action](https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_executor.py#L659) (and optionally, a task can run an action repeatedly using the `loop` keyword, and skip action execution using the `when` keyword). A failed task triggers all sorts of Ansible features (any_errors_fatal, max_fail_percentage, rescue blocks) and a task that skips an action is a successful task that doesn't trigger any failure behaviors or features, it is quite literally a success by the dictionary definition.

**Comment by rptaylor** [comment]
> I [did](https://github.com/ansible/ansible/pull/83705#discussion_r1711751631).

Looks like a disagreement to me. I don't think the way in which a skipped task succeeds (based on no actions performed) is conceptually the same as the manner in which a failed task fails (based on the outcome of the actions performed), but that thread went into the weeds trying to figure out what things mean. 
Anyway if things are not clearly defined in the first place there is no starting point for logic so who knows, anything goes. If "a skipped task is executed" then a "skipped task" is a contradiction in terms and words don't really mean anything anymore. What I understand you to mean is that a skipped task runs but the actions are skipped, but in this inconsistent lexicon the underlying contradiction remains: a skipped task (in Ansible meaning) is not really a skipped task (in English meaning) because there is a sense in which it runs (depending on what you're talking about and how each of those words is interpreted).

> It's ironic since you deleted the line that saves users the effort of looking up the documentation for the failed test in order to understand the success test:

I thought that was a redundant duplication of information since the latest revision linked directly to the failed and skipped tests, but it did not matter much to me either way.

As I mentioned earlier the distinction between tasks and actions does not at first seem particularly material to me (although it does open up other pitfalls). But if it is very material to understanding the skipped, success, and failed tests (e.g. do tests actually test the actions of a task rather than a task itself?) then I guess that is another documentation bug because none of them mention actions.

Anyway the reason I opened this MR is because https://github.com/ansible/ansible/issues/83698 was closed without any other follow up to address or improve the situation. If you already decided what to write I guess you should close this MR and do that. But if this has generated some deeper thought about improving the documentation I think that is a positive outcome.

**Comment by bcoca** [comment]
Actions are skipped, tasks are evaluated, hence a skipped task is still a successful task (not action). A task can fail for several reasons, commonly when it's 'fail condition' is met, by default that is if the action fails. It can also fail due to errors during task evaluation, most commonly invalid inputs or definitions, like when encountering undefined variables.
It gets even more complicated as we add 'unreachable' issues which mostly fail the host and are not considered failures of the task itself nor the action.

Initially, it was not the intent to disclose to the user most of these nuances, but as Ansible grew and we adding things like block/rescue, we did expose these, while the documentation might not be very explicit about these details at times it does correctly state the behaviors.  I think you understand all of this at this point.

Our reviews mostly agree, we might not have expressed this well, but the gist of it is, that we agree that the current text only requires a slight addition, not a rewrite, nor is there a need to change the documentation of the failed test either.  Interpreting each comment with as a contradiction, specially when not meant as such, is not constructive, they were mostly meant to be additive or a parallel way to achieve the same result as there seemed to be a impasse with the previously discussed options.  In the end we all want a clear and concise documentation, but I do not think most of the rewrites in this PR would achieve that. 

I don't mean to be too negative here, but we clearly had a series of miscommunications and have taken this change to a road that seems to have no end. I apologize for the part i played in that, I had not taken enough time to be clear and precise and expected my comments to be enough to convey the full context. I hope we can all be the same page and now move to a resolution.

**Comment by rptaylor** [comment]
@bcoca Thanks for the clarifications and extra context!

> tasks are evaluated

I think that is excellent terminology. The _evaluation_ of tasks results in the _execution_ of actions. With terms like that established it makes it possible to communicate about and distinguish between the dual ways in which tasks operate (internal evaluation on the controller host and external execution on target hosts, though this depends on the nature of specific modules), which in turn makes it possible to articulate the different ways in which a task can be said to succeed, fail, etc. (beyond resorting to the de facto code definition). This is especially important because of the nuance that sometimes it is the status of actions which are ascribed to the task, even though (evaluation of) the task itself does not have that status per se.

> Interpreting each comment with as a contradiction, specially when not meant as such, is not constructive

I appreciate and understand that contradictions are typically not intended, but what is pertinent in the context of writing documentation is whether there is room for contradictions to be interpreted, regardless of intent. My goal was to try to dig into that and uncover what was really meant so that it could be better expressed in the documentation, clarifying terms and resolving ambiguity. My intention in deliberately probing the fuzzy areas was to close that gap, which IMHO would be constructive and valuable. I apologize if it came across differently than intended.

Please accept my regrets that I am no longer able to dedicate any more time to this issue, and as it remains clear that the maintainers know what documentation changes they want to write, and in consideration of what is most productive for everyone, I am going to close this and leave it up to you to proceed with the other MR.  I appreciate your patience and willingness to engage in discussion and I hope it has generated some helpful thoughts about the use of terminology in documentation.

**Pull request closed without merging**
